### ISSUE TITLE
feat: add strict mode (only inject classes annotated with `@Service` decorator)

### ISSUE THREAD

**Issue by tonyxiao** [typestack/typedi#87]
I would like to have a strict mode for `typedi` where unless a class was registered with the `@Service` decorator, it does not try and build an instance of that class. This makes it easier to identify errors (they will be thrown as `ServiceNotFound` error) rather than getting random `undefined` errors later codepath at runtime. Thoughts?

**Comment by NoNameProvided** [comment]
This has a related discussion in #60. The consensus was to 
- strictly require dependencies and throw an error when we find nothing
- add `@Optional` decorator to opt-out from this behavior and inject undefined if an optional operator is not found

```ts
@Service()
constructor(
	@Optional() private myDep: MyDep,
	private myOtherDep: myOtherDep,
) {}
```

My addtion to this that we maybe should add an extra setting to the `@Inject` decorator instead: 

```ts
constructor(
	@Inject({ optional: true }) private myDep: MyDep,
	private myOtherDep: myOtherDep,
) {}
```

**Comment by NoNameProvided** [comment]
This has been implemented in e7144214a3b8c1387010e03e7513d5a2749df846. Will be included in the next release.

**Comment by NoNameProvided** [comment]
This has been released as `0.9.0`.
